resources/providers/azurelib/governance/management_group.go

```diff
@@ -0,0 +1,153 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package governance
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/samber/lo"
+
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib/inventory"
```

Comment:
[Optional/Quesation] Since this package imports the same-layer package inventory and also seems pretty small; why not merge them inside inventory (having a `Provider` and a `GovernanceProvider` ?

Reply:
I feel it is better to start separating things instead of continuously enlarging one single package, it feels that it will go against go [best practices](https://go.dev/blog/package-names).

For example, `governance.Provider` is a better name than `inventory.GovernanceProvider`.

Reply:
In general, I agree, in this case, since they are both "inventory" in a way, might be good enough. I mostly have in mind the [tiny package syndrome](https://about.sourcegraph.com/blog/go/idiomatic-go#tiny-package-syndrome) pitfall.  (_For example we now have another provider `azurelib.Provider` that hides these two providers that they hide the azure SDK calls_). 

Which might be ok. I just got a feeling of a bit of complexity when first looked at it and thus the question. 

Totally optional.